File: undirectedGraph.js
const Graph = require("./graph");

class UndirectedGraph extends Graph {
  constructor() {
    super();
  }

  addEdges(u, v, weight = 0) {
    if (!this.adjList[u]) this.adjList[u] = [];
    if (!this.adjList[v]) this.adjList[v] = [];
    this.adjList[u].push([v, weight]);
    this.adjList[v].push([u, weight]);
  }
}

File: UndirectedGraph.py
from Graph import Graph

class UndirectedGraph(Graph):
  def __init__(self) -> None:
    super().__init__()

  def add_edge(self, u: int, v: int, w: int=0) -> None:
    if u not in self.adj_list: self.adj_list[u] = []
    if v not in self.adj_list: self.adj_list[v] = []
    self.adj_list[u].append([v, w])
    self.adj_list[v].append([u, w])